usr/src/uts/armv8/rpi4/io/bcm2711_pcie/bcm2711_pcie.c

```diff
@@ -269,6 +293,79 @@ bcm2711_pcie_bus_config(dev_info_t *pdip, uint_t flags, ddi_bus_config_op_t op,
 	return (rval);
 }
 
+static uint32_t
+bcm2711_pcie_read_reg(const bcm2711_pcie_softc_t *softc, uint32_t reg)
+{
+	uint32_t val =
+	    ddi_get32(softc->bc_handle, (uint32_t *)(softc->bc_base + reg));
+
+	return (val);
+}
+
+static void
+bcm2711_pcie_write_reg(const bcm2711_pcie_softc_t *softc, uint32_t reg,
+    uint32_t val)
+{
+	ddi_put32(softc->bc_handle, (uint32_t *)(softc->bc_base + reg), val);
+}
+
+static void
+bcm2711_pcie_reset_controller(const bcm2711_pcie_softc_t *softc)
```

Comment:
fewer blank lines? :)

Do we know how BSD handle this?  Do they just know the PCIe is ok because something during boot did this?

Reply:
heh, it seems i like vertical spaces too much. I compacted it, let me know if I shall remove even more blank lines.

The mechanism is taken from FreeBSD. I think it's the safest way, not to assume the controller is in a certain state but reset/enable it explicitly.